File: passenger-service/src/main/java/io/voitovich/yura/passengerservice/exceptionhandler/model/ValidationExceptionInfo.java
package io.voitovich.yura.passengerservice.exceptionhandler.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.NonNull;
import lombok.Singular;
import org.springframework.http.HttpStatus;

import java.util.Map;

@Builder
@Schema(name = "ValidationExceptionInfo", description = "represents all validation exceptions")
public record ValidationExceptionInfo (
    @Schema(name = "status", description = "error status", example = "BAD_REQUEST")
    @NonNull
    HttpStatus status,
    @Schema(name = "errors", description = "validation errors", example = "name: name is mandatory")
    @Singular
    Map<String, String> errors
){}

File: ride-service/src/main/kotlin/io/voitovich/yura/rideservice/exceptionhandler/model/ValidationExceptionInfo.kt
package io.voitovich.yura.rideservice.exceptionhandler.model

import io.swagger.v3.oas.annotations.media.Schema
import org.springframework.http.HttpStatus


@Schema(name = "ValidationExceptionInfo", description = "Represents information about validation exceptions.")
data class ValidationExceptionInfo(

    @Schema(name = "status", description = "Error status", example = "BAD_REQUEST")
    val status: HttpStatus,

    @Schema(name = "errors", description = "Map of validation errors", example = "{'name': 'Name is mandatory'}")
    val errors: MutableMap<String, String?>
)